src/postprocess/chi0.jl

```diff
@@ -3,12 +3,12 @@ using IterativeSolvers
 using ForwardDiff
 using ProgressMeter
 
+# make ldiv! act as a given function
 struct FunctionPreconditioner
-    f_ldiv!
+    f! # f!(y, x) applies f to x and puts it into y
 end
-LinearAlgebra.ldiv!(y, P::FunctionPreconditioner, x) = P.f_ldiv!(y, x)
-LinearAlgebra.ldiv!(P::FunctionPreconditioner, x) = P.f_ldiv!(copy(x), x)
-# (Base.:\)(P::FunctionPreconditioner, x) = P.f_ldiv!(copy(x), x)
+LinearAlgebra.ldiv!(y, P::FunctionPreconditioner, x) = P.f!(y, x)
+LinearAlgebra.ldiv!(P::FunctionPreconditioner, x) = (y=similar(x); P.f!(y, x); x .= y)
```

Comment:
`= (x .= P.f!(similar(x), x)` ? Also I would make `f!(x)` the default, because it is the least-memory paradigm.

Reply:
it really depends on the underlying implementation. Anyway this doesn't matter at all, it's way lower order, so let's revisit this if we need to optimize it

Reply:
I don't mind strongly too be fair. We can leave it as is.